apps/etl-func/src/domain/message-status.ts

```diff
@@ -25,25 +25,52 @@ const statusEnum = z.enum([
 
 const rejectedMessageStatusSchema = z.object({
   fiscalCode: fiscalCodeSchema,
+  id: z.string().min(1),
   isArchived: z.boolean().default(false),
   isRead: z.boolean().default(false),
   messageId: z.string().ulid(),
   rejection_reason: rejectionReasonSchema,
   status: statusEnum.extract(["REJECTED"]),
   updatedAt: z.number(),
+  version: z.number().gte(0),
 });
 
 const notRejectedMessageStatusSchema = z.object({
   fiscalCode: fiscalCodeSchema,
+  id: z.string().min(1),
```

Comment:
TIP: you can use `.ulid()` to validate our IDs

```suggestion
  id: z.string().ulid(),
```

Reply:
the message status id is not an ulid it is something like `${messageId}-000000000000000` where the second part is the version

Reply:
ah 🥲